### PR TITLE
add tdd content

### PR DESCRIPTION
**Pull request by phiros**
- [x] description
- [x] requirements
- [x] resources
  - [x] books
  - [x] web resources
  - [x] videos
- [x] relations
- [x] take your action
  - [x] for programmers new to mob programming
  - [x] next level & things to try out

### CODE REVIEW

File: definitions/content/test-driven_programming.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+When writing tests favour short and easily readable tests. If need be, break up features into easily testable sub-features in order to simplify the test writing process. 
+For each test you should also think what the focus of the test is. If implementation code is required in the test which is not within the focus of the test you should consider to mock or stub this code.    
+
+## Requirements
+
+### Behavior
+* Work in small increments (i.e. prefer small tests and implementation changes)
+* Always write a failing test first which fails for the right reasons (i.e. a functionality is not implemented)
+* Treat test code as documentation: keep it brief but expressive
```
Comment (smoothd) [inline definitions/content/test-driven_programming.md]: I think it's worth pointing out that aspect of TDD explicitly in the description.

File: definitions/content/test-driven_programming.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+tags: [practice]
+---
+# Test-driven development
+
+Why: test coverage, code quality, finding solutions to hard problems, knowledge sharing
+
+Test-driven development is a way of writing code which puts an emphasis on test coverage, correctness, simplicity and knowledge sharing. The basic process behind this technique is: start by writing test code for a feature or a sub-feature which will fail because the implementation does not satisfy this test. Then add only as much implementation code as necessary in order to make the test pass. After that you can write a test for the next feature or sub-feature and repeat the process until all required features are implement. Ideally, this process helps you to never implement a line of code without writing a test for that line first.
+When writing tests favour short and easily readable tests. If need be, break up features into easily testable sub-features in order to simplify the test writing process. 
+For each test you should also think what the focus of the test is. If implementation code is required in the test which is not within the focus of the test you should consider to mock or stub this code.    
+
```
Comment (smoothd) [inline definitions/content/test-driven_programming.md]: Please consider adding another, more subtle aspect of TDD: By writing tests first, it forces you to stay in _problem_ space longer before entering _solution_ space. Thus it helps to understand the problem at hand better.

File: definitions/content/test-driven_programming.md (hunk 1 above)
Comment (phiros) [reply definitions/content/test-driven_programming.md]: done

File: definitions/content/test-driven_programming.md (hunk 2 above)
Comment (phiros) [reply definitions/content/test-driven_programming.md]: done

**Pull request merged**